### ISSUE TITLE
atvscript throws error on same command that works with atvremote

### ISSUE THREAD

**Issue by svdo** [postlund/pyatv#1930]
### Describe the bug

I'm trying to use `pyatv` in this case to control a remote Music app, just like I tend to do with Apple's "Remote" app.

I run `atvremote` as follows (using Fish shell hence the `(cat ...)` notation):

```
$ atvremote --id [deviceid] --dmap-credentials (cat pyatv-dmap-musicserver.creds) playing
  Media type: Unknown
Device state: Playing
       Title: Voulez-Vous
      Artist: ABBA
       Album: Voulez-Vous
       Genre: Pop
    Position: 295/309s (95.5%)
      Repeat: Off
     Shuffle: Off
```

I then run exactly the same command with `atvscript`:

```
$ atvscript --debug --id [deviceid] --dmap-credentials (cat pyatv-dmap-musicserver.creds) playing
{"result": "failure", "datetime": "2023-02-22T14:24:14.918595+01:00", "exception": "app is not supported", "stacktrace": "Traceback (most recent call last):\n  File \"/Users/[redacted]/.asdf/installs/python/3.10.7/lib/python3.10/site-packages/pyatv/scripts/atvscript.py\", line 308, in appstart\n    print(args.output(await _handle_command(args, abort_sem, loop)), flush=True)\n  File \"/Users/[redacted]/.asdf/installs/python/3.10.7/lib/python3.10/site-packages/pyatv/scripts/atvscript.py\", line 204, in _handle_command\n    return await _run_command(atv, args, abort_sem, loop)\n  File \"/Users/[redacted]/.asdf/installs/python/3.10.7/lib/python3.10/site-packages/pyatv/scripts/atvscript.py\", line 211, in _run_command\n    return output_playing(await atv.metadata.playing(), atv.metadata.app)\n  File \"/Users/[redacted]/.asdf/installs/python/3.10.7/lib/python3.10/site-packages/pyatv/support/shield.py\", line 72, in _guard_method\n    return func(self, *args, **kwargs)\n  File \"/Users/[redacted]/.asdf/installs/python/3.10.7/lib/python3.10/site-packages/pyatv/core/facade.py\", line 239, in app\n    return self.relay(\"app\")\n  File \"/Users/[redacted]/.asdf/installs/python/3.10.7/lib/python3.10/site-packages/pyatv/core/relayer.py\", line 90, in relay\n    instance = self._find_instance(\n  File \"/Users/[redacted]/.asdf/installs/python/3.10.7/lib/python3.10/site-packages/pyatv/core/relayer.py\", line 114, in _find_instance\n    raise exceptions.NotSupportedError(f\"{target} is not supported\")\npyatv.exceptions.NotSupportedError: app is not supported\n"}
```

Please note that `dmap` is the only reported protocol when scanning this device.

### Error log

```
2023-02-22 14:37:44 DEBUG [pyatv.scripts]: Running with pyatv 0.10.3
2023-02-22 14:37:44 DEBUG [pyatv.scripts.atvscript]: Started atvscript
2023-02-22 14:37:44 DEBUG [pyatv.support.net]: Binding on *:5353
2023-02-22 14:37:44 DEBUG [pyatv.support.net]: Binding on 127.0.0.1:0
2023-02-22 14:37:44 DEBUG [pyatv.support.net]: Binding on 10.0.42.32:0
2023-02-22 14:37:46 DEBUG [pyatv.core.scan]: Auto-discovered 3C7C98C032BDB5FA at 10.0.42.148:49187 via Protocol.DMAP ({'txtvers': '1', 'dvty': 'Music', 'dvsv': '3331', 'ossi': '0x186A3', 'dbid': 'C06069C8C5260968', 'ctln': 'Music Server', 'icsv': '65540', 'iv': '196624', 'ver': '131077'})
2023-02-22 14:37:46 DEBUG [pyatv.core.scan]: Auto-discovered Music Server_Music at 10.0.42.148:49187 via Protocol.DMAP ({'txtvers': '1', 'dmv': '131085', 'hq': '2326', 'mid': '0xB6810A1839129232', 'version': '196621', 'itsh version': '196624', 'dvsv': '3331', 'dvty': 'Music', 'machine name': 'Music Server', 'ossi': '0x186A3', 'icsv': '65540', 'hg': '00000000-0248-0404-32ac-5fb3ba58433f', 'database id': 'C06069C8C5260968', 'machine id': '2A76C81C9B03', 'hc': 'bdeedfb1-974f-42ae-bdab-292052f03bff'})
2023-02-22 14:37:46 DEBUG [pyatv.core.facade]: Adding handler for protocol Protocol.DMAP
2023-02-22 14:37:46 DEBUG [pyatv.core.facade]: Connecting to protocol: Protocol.DMAP
2023-02-22 14:37:46 DEBUG [pyatv.support.http]: GET URL: http://10.0.42.148:49187/login?hsgid=00000000-0248-0404-32ac-5fb3ba58433f&hasFP=1
2023-02-22 14:37:46 DEBUG [pyatv.support.http]: Response: status=200, headers=[<CIMultiDictProxy('Date': 'Wed, 22 Feb 2023 13:37:45 GMT', 'DAAP-Server': 'iTunes/13.3.3.2 (OS X)', 'Content-Type': 'application/x-dmap-tagged', 'Content-Length': '47', 'Content-Encoding': 'gzip')>]
2023-02-22 14:37:46 DEBUG [pyatv.support.http]: << GET (Data=6d6c6f67000000186d73747400000004000000c86d6c696400000004422ab58a)
2023-02-22 14:37:46 DEBUG [pyatv.protocols.dmap.daap]: mlog: [container, dmap.loginresponse]
  mstt: 200 [uint, dmap.status]
  mlid: 1110095242 [uint, dmap.sessionid]

2023-02-22 14:37:46 INFO [pyatv.protocols.dmap.daap]: Logged in and got session id 1110095242
2023-02-22 14:37:46 DEBUG [pyatv.protocols.dmap.daap]: Already logged in, re-using seasion id 1110095242
2023-02-22 14:37:46 DEBUG [pyatv.support.http]: GET URL: http://10.0.42.148:49187/ctrl-int/1/playstatusupdate?session-id=1110095242&revision-number=0
2023-02-22 14:37:46 DEBUG [pyatv.support.http]: Response: status=200, headers=[<CIMultiDictProxy('Date': 'Wed, 22 Feb 2023 13:37:45 GMT', 'DAAP-Server': 'iTunes/13.3.3.2 (OS X)', 'Content-Type': 'application/x-dmap-tagged', 'Content-Length': '249', 'Content-Encoding': 'gzip')>]
2023-02-22 14:37:46 DEBUG [pyatv.support.http]: << GET (Data=636d7374000001716d73747400000004000000c8636d7372000000040000004163617073000000010463617368000000010063617270000000010063616673000000010063617673000000010063617663000000010163616173000000040000000263616172000000040000000663616665000000010063617665000000010163616e70000000100000004000000000000000000000217a63616e6e0000001b546865204b696e6720486173204c6f7374204869732043726f776e63616e61000000044142424163616e6c0000000b566f756c657a2d566f757363616e6700000003506f706173616900000008258985c1f628651a636d6d6b00000004000...)
2023-02-22 14:37:46 DEBUG [pyatv.protocols.dmap.daap]: cmst: [container, dmcp.playstatus]
  mstt: 200 [uint, dmap.status]
  cmsr: 65 [uint, dmcp.serverrevision]
  caps: 4 [uint, dacp.playstatus]
  cash: 0 [uint, dacp.shufflestate]
  carp: 0 [uint, dacp.repeatstate]
  cafs: 0 [uint, dacp.fullscreen]
  cavs: 0 [uint, dacp.visualizer]
  cavc: True [bool, dacp.volumecontrollable]
  caas: 2 [uint, dacp.albumshuffle]
  caar: 6 [uint, dacp.albumrepeat]
  cafe: False [bool, dacp.fullscreenenabled]
  cave: True [bool, dacp.dacpvisualizerenabled]
  canp: 0x0000004000000000000000000000217a [bytes, daap.nowplayingid]
  cann: The King Has Lost His Crown [str, daap.nowplayingtrack]
  cana: ABBA [str, daap.nowplayingartist]
  canl: Voulez-Vous [str, daap.nowplayingalbum]
  cang: Pop [str, dacp.nowplayinggenre]
  asai: 2704840119317718298 [uint, daap.songalbumid]
  cmmk: 1 [uint, dmcp.mediakind]
  aeGs: True [bool, com.apple.itunes.can-be-genius-seed]
  ceGS:  [str, com.apple.itunes.genius-selectable]
  casa: 1 [uint, unknown tag]
  aels: 0 [uint, com.apple.itunes.liked-state]
  aelb: False [bool, com.apple.itunes.like-button]
  astm: 213784 [uint, daap.songtime]
  casc: 1 [uint, unknown tag]
  caks: 6 [uint, unknown tag]
  cant: 109020 [uint, dacp.remainingtime]
  cast: 213784 [uint, dacp.tracklength]
  casu: 1 [uint, dacp.su]
  ceQu: False [bool, unknown tag]

2023-02-22 14:37:46 DEBUG [pyatv.core.facade]: Connected to protocol: Protocol.DMAP
2023-02-22 14:37:46 DEBUG [pyatv.core.facade]: Power management not supported by any protocols
2023-02-22 14:37:46 DEBUG [pyatv.protocols.dmap.daap]: Already logged in, re-using seasion id 1110095242
2023-02-22 14:37:46 DEBUG [pyatv.support.http]: GET URL: http://10.0.42.148:49187/ctrl-int/1/playstatusupdate?session-id=1110095242&revision-number=0
2023-02-22 14:37:46 DEBUG [pyatv.support.http]: Response: status=200, headers=[<CIMultiDictProxy('Date': 'Wed, 22 Feb 2023 13:37:45 GMT', 'DAAP-Server': 'iTunes/13.3.3.2 (OS X)', 'Content-Type': 'application/x-dmap-tagged', 'Content-Length': '249', 'Content-Encoding': 'gzip')>]
2023-02-22 14:37:46 DEBUG [pyatv.support.http]: << GET (Data=636d7374000001716d73747400000004000000c8636d7372000000040000004163617073000000010463617368000000010063617270000000010063616673000000010063617673000000010063617663000000010163616173000000040000000263616172000000040000000663616665000000010063617665000000010163616e70000000100000004000000000000000000000217a63616e6e0000001b546865204b696e6720486173204c6f7374204869732043726f776e63616e61000000044142424163616e6c0000000b566f756c657a2d566f757363616e6700000003506f706173616900000008258985c1f628651a636d6d6b00000004000...)
2023-02-22 14:37:46 DEBUG [pyatv.protocols.dmap.daap]: cmst: [container, dmcp.playstatus]
  mstt: 200 [uint, dmap.status]
  cmsr: 65 [uint, dmcp.serverrevision]
  caps: 4 [uint, dacp.playstatus]
  cash: 0 [uint, dacp.shufflestate]
  carp: 0 [uint, dacp.repeatstate]
  cafs: 0 [uint, dacp.fullscreen]
  cavs: 0 [uint, dacp.visualizer]
  cavc: True [bool, dacp.volumecontrollable]
  caas: 2 [uint, dacp.albumshuffle]
  caar: 6 [uint, dacp.albumrepeat]
  cafe: False [bool, dacp.fullscreenenabled]
  cave: True [bool, dacp.dacpvisualizerenabled]
  canp: 0x0000004000000000000000000000217a [bytes, daap.nowplayingid]
  cann: The King Has Lost His Crown [str, daap.nowplayingtrack]
  cana: ABBA [str, daap.nowplayingartist]
  canl: Voulez-Vous [str, daap.nowplayingalbum]
  cang: Pop [str, dacp.nowplayinggenre]
  asai: 2704840119317718298 [uint, daap.songalbumid]
  cmmk: 1 [uint, dmcp.mediakind]
  aeGs: True [bool, com.apple.itunes.can-be-genius-seed]
  ceGS:  [str, com.apple.itunes.genius-selectable]
  casa: 1 [uint, unknown tag]
  aels: 0 [uint, com.apple.itunes.liked-state]
  aelb: False [bool, com.apple.itunes.like-button]
  astm: 213784 [uint, daap.songtime]
  casc: 1 [uint, unknown tag]
  caks: 6 [uint, unknown tag]
  cant: 109020 [uint, dacp.remainingtime]
  cast: 213784 [uint, dacp.tracklength]
  casu: 1 [uint, dacp.su]
  ceQu: False [bool, unknown tag]
```

### How to reproduce the bug?

Open Music app on macOS and run the commands as described above.

### What is expected behavior?

JSON equivalent of the `atvremote` output should be printed.

### Operating System

macOS

### Python

3.10

### pyatv

0.10.3

### Device

Remote iTunes library

### Additional context

MacOS version 13.2.1, Music version 1.3.3.2.

**Comment by svdo** [comment]
Not sure if this is in any way related to #1502.

**Comment by postlund** [comment]
Ah, right. I need to check that the `App` feature is supported. I will try to fix that.

**Comment by postlund** [comment]
Should be fixed by #1962.
